server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -86,13 +88,17 @@ private static NodeUsageStatsForThreadPools.ThreadPoolUsageStats replaceWritePoo
     ) {
         final NodeUsageStatsForThreadPools.ThreadPoolUsageStats writeThreadPoolStats = value.threadPoolUsageStatsMap()
             .get(ThreadPool.Names.WRITE);
-        return new NodeUsageStatsForThreadPools.ThreadPoolUsageStats(
-            writeThreadPoolStats.totalThreadPoolThreads(),
-            (float) Math.max(
-                (writeThreadPoolStats.averageThreadPoolUtilization() + (writeLoadDelta / writeThreadPoolStats.totalThreadPoolThreads())),
-                0.0
-            ),
-            writeThreadPoolStats.averageThreadPoolQueueLatencyMillis()
+        final float newWritePoolUtilization = (float) Math.max(
+            (writeThreadPoolStats.utilizationSamples().getLast().utilization() + (writeLoadDelta / writeThreadPoolStats.numberOfThreads())),
+            0.0
         );
+        final List<NodeUsageStatsForThreadPools.UtilizationSample> newUtilizationSamples = new ArrayList<>(
+            writeThreadPoolStats.utilizationSamples()
+        );
+        final NodeUsageStatsForThreadPools.UtilizationSample previousUtilization = newUtilizationSamples.removeLast();
+        newUtilizationSamples.add(
+            new NodeUsageStatsForThreadPools.UtilizationSample(previousUtilization.instant(), newWritePoolUtilization)
+        );
+        return new NodeUsageStatsForThreadPools.ThreadPoolUsageStats(writeThreadPoolStats.numberOfThreads(), newUtilizationSamples);
```

Comment:
Simulator will replace the most recent utilization value with the new one. We could instead add one to the end, but then what timestamp would we put on it? 🤷 

Hard to know the best strategy without knowing how the determination of "hot spotting" is made, and whether we care about that in the simulator.